### PR TITLE
feat(outfitter): add migrate kit codemod command

### PR DESCRIPTION
**Pull request by galligan**
## Summary
- Adds `outfitter migrate-kit` codemod command for adoption/migration.
- Performs AST-based import rewrites for kit migration scenarios.

## Scope
- Codemod command implementation and migration transforms.
- Tests for rewrite correctness and edge cases.

## Validation
- `bun run typecheck`
- `bun run check`
- `bun run build`
- `bun run test`

### CODE REVIEW

File: apps/outfitter/src/actions.ts [hunk 1]
```diff
@@ -90,6 +100,16 @@ const createInputSchema = z.object({
...
 
+interface MigrateKitActionInput extends MigrateKitOptions {
+  outputMode?: OutputMode;
+}
+
+const migrateKitInputSchema = z.object({
+  targetDir: z.string(),
+  dryRun: z.boolean(),
+  outputMode: outputModeSchema,
+}) as z.ZodType<MigrateKitActionInput>;
```
[the remark this answers is not in the document]
Comment (galligan) [reply apps/outfitter/src/actions.ts]: Addressed. Migrate flag mapping now normalizes Commander forms (`dryRun` and `dry-run`) into typed action input so the schema path is reachable in practice.

File: apps/outfitter/src/commands/migrate-kit.ts [hunk 2]
```diff
@@ -0,0 +1,955 @@
...
+  interface CliFlags {
+    readonly dryRun?: unknown;
+    readonly json?: unknown;
+  }
+
+  const migrate = program.command("migrate").description("Migration commands");
+
+  migrate
+    .command("kit [directory]")
+    .description(
```
[the remark this answers is not in the document]
Comment (galligan) [reply apps/outfitter/src/commands/migrate-kit.ts]: Addressed. `migrateKitCommand` now reuses an existing `migrate` group and no-ops if `kit` is already registered, preventing duplicate registration.

File: apps/outfitter/src/commands/migrate-kit.ts [hunk 3]
```diff
@@ -0,0 +1,955 @@
...
+
+  while (stack.length > 0) {
+    const currentDir = stack.pop();
+    if (!currentDir) continue;
+
+    const entries = readdirSync(currentDir).sort((a, b) => a.localeCompare(b));
+
+    for (const entry of entries) {
+      const fullPath = join(currentDir, entry);
+      const stat = statSync(fullPath);
```
[the remark this answers is not in the document]
Comment (galligan) [reply apps/outfitter/src/commands/migrate-kit.ts]: Addressed. Source traversal now catches `readdirSync`/`statSync` failures and returns `Result.err(MigrateKitError)` instead of throwing.
